src/main/java/org/arl/fjage/connectors/WebServer.java

```diff
@@ -443,9 +450,18 @@ public void handle(String target, Request baseRequest, HttpServletRequest reques
           filename = URLEncoder.encode(filename, String.valueOf(StandardCharsets.UTF_8));
           Path outputFile = outputDir.resolve(filename);
           try (InputStream inputStream = part.getInputStream();
-               OutputStream outputStream = Files.newOutputStream(outputFile, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)) {
+               FileOutputStream outputStream = new FileOutputStream(outputFile.toFile())) {
             IO.copy(inputStream, outputStream);
+            // Force data and metadata to disk
+            outputStream.getChannel().force(true);
+            // Ensure lowest-level disk synchronization
+            outputStream.getFD().sync();
             out.printf("%s%n", outputFile);
+          }catch (Throwable ex){
+            log.warning("Unable to save file : " + filename + " : " + ex.toString());
```

Comment:
I know there's other lines analogous to this one in this file, but can we do
```suggestion
            log.log(Level.WARNING, "Unable to save file : " + filename, ex);
```
(requires importing `java.util.logging.Level`)? This prints the whole stack trace, while your version prints only the error message.

Reply:
Right, we probably want to go about and do this in most other places where we log an exception.